src/Arcus.Messaging.ServiceBus.Core/ServiceBusMessageBuilder.cs

```diff
@@ -0,0 +1,156 @@
+
+        /// <summary>
+        /// Adds a <paramref name="operationParentId"/> as an application property in <see cref="ServiceBusMessage.ApplicationProperties"/>
+        /// with <see cref="PropertyNames.OperationParentId"/> as key.
+        /// </summary>
+        /// <param name="operationParentId">The unique identifier of the current parent operation.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="operationParentId"/> is blank.</exception>
+        public ServiceBusMessageBuilder WithOperationParentId(string operationParentId)
+        {
+            Guard.NotNullOrWhitespace(operationParentId, nameof(operationParentId), "Requires a non-blank transaction ID for the Azure Service Bus message");
+            return WithOperationParentId(operationParentId, PropertyNames.OperationParentId);
+        }
+
+        /// <summary>
+        /// Adds a <paramref name="operationParentId"/> as an application property in <see cref="ServiceBusMessage.ApplicationProperties"/>
+        /// with <paramref name="operationParentIdPropertyName"/> as key.
+        /// </summary>
+        /// <param name="operationParentId">The unique identifier of the current operation.</param>
+        /// <param name="operationParentIdPropertyName">The name of the application property for the <paramref name="operationParentId"/>.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="operationParentId"/> or the <paramref name="operationParentIdPropertyName"/> is blank.</exception>
+        public ServiceBusMessageBuilder WithOperationParentId(
+            string operationParentId,
+            string operationParentIdPropertyName)
+        {
+            Guard.NotNullOrWhitespace(operationParentId, nameof(operationParentId), "Requires a non-blank transaction ID for the Azure Service Bus message");
+            Guard.NotNullOrWhitespace(operationParentIdPropertyName, nameof(operationParentIdPropertyName), "Requires a non-blank property name of the transaction ID for the Azure Service Bus message");
+
+            _operationParentIdProperty = new KeyValuePair<string, object>(operationParentIdPropertyName, operationParentId);
+            return this;
+        }
+
+        /// <summary>
+        /// Creates an <see cref="ServiceBusMessage"/> instance based on the configured settings.
+        /// </summary>
+        public ServiceBusMessage Build()
+        {
+            string json = JsonConvert.SerializeObject(_messageBody);
+            byte[] raw = _encoding.GetBytes(json);
+            var message = new ServiceBusMessage(raw)
+            {
+                ApplicationProperties =
+                {
+                    { PropertyNames.ContentType, "application/json" },
+                    { PropertyNames.Encoding, _encoding.WebName }
+                }
+            };
+
+            if (_operationId != null)
+            {
+                message.CorrelationId = _operationId;
+            }
```

Comment:
Do we have this standard behavior somewhere documented ?

I'm not saying this is wrong; just wondering how we decided that the operationId should be used to set the CorrelationId, and why for instance we do not use the transactionId for that ?

I was just reading up on [this](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messages-payloads#message-routing-and-correlation), hence my question

Reply:
In our docs, yeah, maybe not that explicitly. This functionality was previously in the `ObjectExtension` but it felt like that extension was becoming to big and complex to be readable, so this message build is a alternative. When I read it correctly in the Microsoft docs, the `CorrelationId` is closely linked to the message itself, which is the exact purpose of the operation ID, and not the transaction ID. It will be a good idea, indeed, to more explicitly document this message builder and how the correlation is set.

Reply:
Fine for me.  I was just wondering if there should be a way for users to customize / override this behavior.

Reply:
Yeah, that's definitely a good point. Maybe we can provide an overload where they can provide an property name, so in that case the operation ID is (like the transaction ID and operation parent ID) assigned to the application properties of the message. Would that be ok?

Reply:
I'll rename the method to `WithOperationId` and created a follow-up issue: #279 .